src/socket.rs

```diff
@@ -655,6 +655,7 @@ fn set_common_type(ty: Type) -> Type {
 
 /// Set `FD_CLOEXEC` and `NOSIGPIPE` on the `socket` for platforms that need it.
 #[inline(always)]
+#[allow(clippy::unnecessary_wraps)]
```

Comment:
This doesn't seem necessary.

Reply:
It prevents that Clippy warning from firing on platforms where the function is a no-op. Should I remove it?

Reply:
That's why it didn't trigger on macOS. In that case you can leave it.